src/Magento/FunctionalTestingFramework/Module/MagentoActionProxies.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\Module;
+
+use Codeception\Module as CodeceptionModule;
+use Magento\FunctionalTestingFramework\DataGenerator\Handlers\PersistedObjectHandler;
+use Magento\FunctionalTestingFramework\DataGenerator\Handlers\CredentialStore;
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+
+/**
+ * Class MagentoActionProxies
+ *
+ * Contains all proxy functions whose corresponding MFTF actions need to be accessible for AcceptanceTester $I
+ *
+ * @package Magento\FunctionalTestingFramework\Module
+ */
+class MagentoActionProxies extends CodeceptionModule
+{
+    /**
+     * PersistedObjectHandler instance
+     *
+     * @var PersistedObjectHandler
+     */
+    private static $persistHandler = null;
```

Comment:
Storing a reference to a singleton does not play into the advantages of the design pattern; the whole point of it is that any time we want to reference the singleton we can `Singleton::getInstance()` without instantiating it (or having to keep a reference to an instantiated singleton.

I would strongly reccomend removal of this private static variable, as well as removing all of the `if (!self::persistHandler)` clauses. This should work to keep this file even more trim than it already is.

I have the feeling I'm missing some partial context here, so please feel free to enlighten me if I'm missing a key reason for this design decision.

Reply:
I am aware of the singleton nature of PersistedObjectHandler, and was trying to make it more explicit. I guess it's not necessary. :)

Removed.